.github/workflows/docs.yml

```diff
@@ -0,0 +1,43 @@
+name: Documentation
+
+on:
+  [push, pull_request]
+
+jobs:
+  build:
+    name: Build HTML documentation
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-python@v2
+        name: Install Python
+        with:
+          python-version: '3.9'
+
+      - name: Install documentation dependencies
+        run: |
+          python -mpip install -r doc/requirements.txt
```

Comment:
We don't distribute the `doc` folder in the sdist or wheel of the package, so including the requirements there would be unnecessary.  It also means that the `[full]` target would have a lot of very hard, fixed dependencies, which would make package resolution very difficult.  `requirements.txt` is generally for fixing a single, working, reproducible build, rather than showing the minimum constraints, which is what the `setup.cfg` targets do.

Particularly with the documentation build, we have quite an interest in making sure the output is _exactly_ as we want it to be - there are various inconsistencies between different versions of `docutils` and `sphinx_rtd_theme`, and things like that, and we don't want the layout to change from build to build.

Reply:
Ok, I see. Thank you for explaining it!